src/sql-parser/src/parser.rs

```diff
@@ -101,6 +101,21 @@ pub struct Parser {
     index: usize,
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub enum Precedence {
+    Zero,
```

Comment:
Perhaps throw an "order controls the precedence" comment in here? It's easy to read past the `Ord` derivation.